R/emptyTheseVariables.R

```diff
@@ -0,0 +1,32 @@
+
+#############################################################################
+#' Set variables to \code{NA}.
+#'
+#' Set all values within one or multiple variables to \code{NA}.
+#'
+#'@param GADSdat A \code{GADSdat} object.
+#'@param vars Character vector of variable names which should be set to \code{NA}.
+#'
+#'@return Returns the recoded \code{GADSdat}.
+#'
+#'@examples
+#' # empty multiple variables
+#' pisa2 <- emptyTheseVariables(pisa, vars = c("idstud", "idschool"))
+#'
+#'
+#'@export
+emptyTheseVariables <- function(GADSdat, vars) {
+  UseMethod("emptyTheseVariables")
+}
+
+#'@export
+emptyTheseVariables.GADSdat <- function(GADSdat, vars) {
+  check_GADSdat(GADSdat)
+  if(!is.character(vars) || length(vars) < 1) stop("'vars' needs to be character vector of at least length 1.")
+  check_vars_in_GADSdat(GADSdat, vars = vars)
+
+  for(nam in vars) {
+    GADSdat$dat[[nam]][] <- NA
```

Comment:
In fact the last square bracket is vital, as it preserves the vector type (e.g., numeric vectors stay numeric, character vectors stay character). Otherwise, all vectors are transformed to logical (as `NA` is logical). This would lead to incompatabilities with the spss format meta information.

Reply:
Ah, I see - thank you for explanation! I was not aware of this "trick". Then it's perfectly fine, of course!